src/main/java/ch/njol/skript/entity/FallingBlockData.java

```diff
@@ -122,75 +129,71 @@ public FallingBlock spawn(Location loc, @Nullable Consumer<FallingBlock> consume
 			assert false : t;
 			return null;
 		}
-		FallingBlock fallingBlock = loc.getWorld().spawnFallingBlock(loc, material.createBlockData());
+		FallingBlock fallingBlock = loc.getWorld().spawnFallingBlock(loc, data == null ? material.createBlockData() : data);
 		if (consumer != null)
 			consumer.accept(fallingBlock);
 
 		return fallingBlock;
 	}
 
 	@Override
-	public void set(final FallingBlock entity) {
+	public void set(FallingBlock entity) {
 		assert false;
 	}
-	
+
 	@Override
 	public Class<? extends FallingBlock> getType() {
 		return FallingBlock.class;
 	}
-	
+
 	@Override
-	public boolean isSupertypeOf(final EntityData<?> e) {
+	public boolean isSupertypeOf(EntityData<?> e) {
 		if (!(e instanceof FallingBlockData))
 			return false;
-		final FallingBlockData d = (FallingBlockData) e;
+		FallingBlockData d = (FallingBlockData) e;
 		if (types != null) {
 			if (d.types != null)
 				return ItemType.isSubset(types, d.types);
 			return false;
 		}
 		return true;
 	}
-	
+
 	@Override
 	public EntityData getSuperType() {
 		return new FallingBlockData(types);
 	}
-	
+
 	@Override
-	public String toString(final int flags) {
-		final ItemType[] types = this.types;
+	public String toString(int flags) {
 		if (types == null)
 			return super.toString(flags);
-		final StringBuilder b = new StringBuilder();
-		b.append(Noun.getArticleWithSpace(types[0].getTypes().get(0).getGender(), flags));
-		b.append(m_adjective.toString(types[0].getTypes().get(0).getGender(), flags));
-		b.append(" ");
-		b.append(Classes.toString(types, flags & Language.NO_ARTICLE_MASK, false));
-		return "" + b.toString();
+		StringBuilder builder = new StringBuilder();
+		builder.append(Noun.getArticleWithSpace(types[0].getTypes().get(0).getGender(), flags));
+		builder.append(m_adjective.toString(types[0].getTypes().get(0).getGender(), flags));
+		builder.append(" ");
+		builder.append(Classes.toString(types, flags & Language.NO_ARTICLE_MASK, false));
+		return builder.toString();
 	}
-	
-//		return ItemType.serialize(types);
+
 	@Override
 	@Deprecated
-	protected boolean deserialize(final String s) {
+	protected boolean deserialize(String s) {
 		throw new UnsupportedOperationException("old serialization is not supported");
-//		if (s.isEmpty())
-//			return true;
-//		types = ItemType.deserialize(s);
-//		return types != null;
 	}
-	
+
 	@Override
-	protected boolean equals_i(final EntityData<?> obj) {
+	protected boolean equals_i(EntityData<?> obj) {
 		if (!(obj instanceof FallingBlockData))
 			return false;
+		if (data != null && ((FallingBlockData) obj).data != null)
+			return data.equals(((FallingBlockData) obj).data);
```

Comment:
Can block data checking replace item type checking? if not it shouldn't return if true only if false